preprocessy/input/_read.py

```diff
@@ -2,13 +2,20 @@
 import os
 import errno
 
+
 class ReadData(object):
     def __init__(self, file_name):
         self.excel_extensions = ["xls", "xlsx", "xlsm", "xlsb", "odf", "ods", "odt"]
-        self.file_name = file_name
+        self._validate_input(file_name)
         self.__read_file()
         self.summary, self.stats = self.__read_summary()
 
+    def _validate_input(self, file_name):
+        if type(file_name) is not str:
+            raise TypeError("Argument file_name should be of str type")
```

Comment:
Can you also show what was the type of `file_name` received? So the error would look like

> Argument file_name should be of type str. Received xyz

You can use `f strings` for this.

Reply:
Yes will do the changes